### ISSUE TITLE
Empty /proc/<pid>/cmdline may cause string index out of bound problem

### ISSUE THREAD

**Issue by duanyao** [raboof/nethogs#144]
Some processes in Linux has empty `/proc/<pid>/cmdline`, so in the following code,

https://github.com/raboof/nethogs/blob/7093964413c086f7eee06cbe257f2a579caae209/src/inode2prog.cpp#L129

because `cmdline.length()` is unsigned, `cmdline.length() - 1` is a very large number, this loop would run out of bound can cause a SIGEV.

This does happen frequently in my box.

**Comment by raboof** [comment]
Interesting! Would you be interested in contributing a fix?

**Comment by duanyao** [comment]
I have found that a [zombie process](https://en.wikipedia.org/wiki/Zombie_process) always has empty  `/proc/<pid>/cmdline`, so if there are many short-lived processes in a system, nethogs will easily hit this problem. E.g. this node.js script do this:

```js
var child_process = require('child_process');

function run() {
    child_process.exec(
        'curl http://localhost:3000', (error, stdout, stderr) => { }
    );
    setTimeout(run, 10);
}

run();
```

**Comment by raboof** [comment]
reportedly fixed by #175
